dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java

```diff
@@ -498,6 +498,7 @@ public final Response logoutAs(@Context final HttpServletRequest httpServletRequ
 			final Map<String, Object> sessionData = this.helper.doLogoutAs(principalUserId, currentLoginAsUser, serverName);
 			revertLoginAsSessionInfo(httpServletRequest, Host.class.cast(sessionData.get(com.dotmarketing.util.WebKeys
 					.CURRENT_HOST)), principalUserId);
+			httpServletRequest.getSession().removeAttribute(WebKeys.PRINCIPAL_USER_ID);
```

Comment:
Move this code to the `revertLoginAsSessionInfo` method in order to encapsulate the session revert process

Reply:
done